fiftyone/core/collections.py

```
@@ -10521,21 +10604,28 @@ def aggregate(self, aggregations, _mongo=False):
             return pipelines[0] if scalar_result else pipelines
 
         # Run all aggregations
-        _results = foo.aggregate(self._dataset._sample_collection, pipelines)
+        _results = foo.aggregate(
+            self._dataset._sample_collection, pipelines, _stream=stream
+        )
 
         # Parse batch results
         if batch_aggs:
+            if stream:
+                return self._iter_and_parse_agg_results(batch_aggs, _results)
```

Comment on code readability: the early return here is messy. The flow of the `aggregate(aggs)` method is to break up the aggregations into categories, get their results, and insert them into `results` in the correct positions.

What would be *ideal* is to make `aggregate()` support a mix of lazy and non-lazy aggregations. Just like how the current implementation supports a mix of "big", "batchable", and "other" aggregations. It treats them appropriately behind the scenes but the user can still pass an arbitrary list of things and we'll get them done.

However, I guess we cannot do that as "stream" aggregations are lumped together in a single generator that emits `(result1, result2, ...)` tuples, which is a fundamentally different return format than what `aggregate()` provides for non-lazy results.

So, now I'm thinking we should raise an error if a mix of lazy and non-lazy aggregations are provided.